include/display.h

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2017 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for display drivers and applications
+ */
+
+#ifndef __DISPLAY_H__
+#define __DISPLAY_H__
+
+/**
+ * @brief Display Interface
+ * @defgroup display_interface display Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <device.h>
+#include <stddef.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum display_pixel_format {
+	PIXEL_FORMAT_RGB_888 = (1 << 0),
+};
+
+/**
+ * @struct display_capabilities
+ * @brief Structure holding display capabilities
+ *
+ * @var display_capabilities::x_resolution
+ * Display resolution in the X direction
+ *
+ * @var display_capabilities::y_resolution
+ * Display resolution in the Y direction
+ *
+ * @var display_capabilities::supported_pixel_formats
+ * Bitwise or of pixel formats supported by the display
+ */
+struct display_capabilities {
+	u16_t x_resolution;
+	u16_t y_resolution;
+	enum display_pixel_format supported_pixel_formats;
```

Comment:
If this is mask, then u32_t will be better than enum, which is intended to hold only for single value from set.